File: Problem-034/solution.js
function digitFactorial() {
    let total = 0;
    let numbers = [];

    for (let num = 10; num <= 7 * fac(9); num++) {
        let sum = 0;
        for (let digit of num.toString()) {
            sum += fac(Number(digit));
        }

        if (sum === num) {
            total += num;
            numbers.push(num);
        }
    }
    return {
        sum: total,
        numbers: numbers
    };
}


function fac(n) {
    let prod = 1;

    for (let i = 1; i <= n; i++) {
        prod *= i;
    }

    return prod;
}

File: Problem-034/solution.py
def factorial(n):
    result = 1

    for i in range(1, n + 1):
        result *= i

    return result


def digit_factorial():
    total = 0

    for num in range(10, 2540161):
        digit_sum = 0

        for digit in str(num):
            digit_sum += factorial(int(digit))

        if digit_sum == num:
            total += num

    return total


print(digit_factorial())
